File: Clue/GameObject.py
#!/usr/bin/python

class GameObject:
    _id = 0;
    _label = "default label"

    def __init__(self, ID, label):
        self._id = ID
        self._label = label
        
    def __str__(self):
        return ''.join((
            "id: ",
            self._id,
            " | ",
            self._label
        ))

    def getId(self):
        return self._id

    def getLabel(self):
        return self._label

File: Java/src/main/java/clue/logic/GameObject.java
package clue.logic;

import javax.persistence.*;

//@Entity
////@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
//@DiscriminatorColumn(
//    name = "OBJECT_TYPE",
//    discriminatorType = DiscriminatorType.STRING)
@MappedSuperclass
public class GameObject {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", updatable = false, nullable = false)
  protected long id;
  @Column(insertable = false, updatable = false)
  protected String label;

  public GameObject() {
    label = "NOT SET";
  }

  public GameObject(String aLabel) {
    label = aLabel;
  }

  public GameObject(long id, String label) {
    this.id = id;
    this.label = label;
  }

  public long getId() {
    return id;
  }

  public String getLabel() {
    return label;
  }

  public void setLabel(String label) {
    this.label = label;
  }

  public String toString() {

    return id + " | " + label;
  }
}
